### PR TITLE
Abandon starter-kit and build environment with webpack

### PR DESCRIPTION
**Pull request by ss77995ss**
# Description
* Install `webpack` modules and use it to build `production` & `development` environment
* Implement `uglify` & `minify` in `production` environment
* Run `dev-server` in `development environment`

# How to test
* Use shell command `npm run start` & `npm run build` to test `development` & `production` environment setup

# Questions
* How to split bundle more properly
* When I use devtool in browser, I only can see `app.bundle.js` file and cannot see other files in the project

### CODE REVIEW

File: webpack.prod.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const merge = require('webpack-merge');
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const common = require('./webpack.common.js');
+
+module.exports = merge(common, {
+  mode: 'production',
+  optimization: {
```
Comment (chunming-c) [inline webpack.prod.js]: Plugins for `production` mode should contain `minimization` already. Do we really need this?

ref: https://medium.com/webpack/webpack-4-mode-and-optimization-5423a6bc597a

File: webpack.common.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+const htmlPlugin = new HtmlWebPackPlugin({
+  template: 'src/client/index.html',
+  filename: 'index.html',
+});
+
+module.exports = {
+  entry: {
+    app: './src/client/index.js',
+    actions: './src/client/actions/index.js',
```
Comment (chunming-c) [inline webpack.common.js]: Good!

File: package.json [hunk 3]
```diff
@@ -3,11 +3,11 @@
...
   "license": "MIT",
   "scripts": {
-    "start": "razzle start",
-    "build": "razzle build",
+    "start": "webpack-dev-server --open --config webpack.dev.js",
+    "build": "webpack --config webpack.prod.js",
     "test": "jest",
     "report": "jest --coverage",
-    "start:prod": "NODE_ENV=production node build/server.js",
+    "server": "babel-node src/index.js",
```
Comment (chunming-c) [inline package.json]: Are you meaning `server` to start the server for production? If yes, `babel-node` is not the correct way. Please take look on the [documentation](https://babeljs.io/docs/en/babel-node#not-meant-for-production-use).

File: .babelrc [hunk 4]
```diff
@@ -1,10 +1,11 @@
 {
   "presets": [
-    ["env"],
-    "react"
+    ["@babel/preset-env"],
```
Comment (chunming-c) [inline .babelrc]: `[]` seems redundant here.

File: src/client/index.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
```
Comment (chunming-c) [inline src/client/index.html]: Instead of leaving your `index.html` here, I suggest to reposition to in the `public` folder under the root.

```diff
root
  ├── 
  ├── ...
  ├── webpack.dev.js
  ├── webpack.prod.js
  └── public
+     └── index.html
```

**Review by chunming-c — COMMENTED** [review]
Looks better. Keep moving on.

File: webpack.prod.js (hunk 1 above)
Comment (ss77995ss) [reply webpack.prod.js]: I think maybe I can customize `minimization`, but maybe default config is enough. I will read the document and figure it out.

File: package.json (hunk 3 above)
Comment (ss77995ss) [reply package.json]: I tried to use `src/index.js` files to run development like previous build, but not succeeded. And I forgot to erase this script.

File: src/client/index.html (hunk 5 above)
Comment (ss77995ss) [reply src/client/index.html]: OK, this case also remind me that in the previous build, `razzle start` can generate a `index.html` file by itself, so maybe we can also create a file to do this as well.

**Review by chunming-c — APPROVED** [review]

**Pull request merged**